odoo/addons/base/tests/test_res_company.py

```diff
@@ -50,3 +50,41 @@ def test_logo_check(self):
         self.assertTrue(company.uses_default_logo)
         company.partner_id.image_1920 = "R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
         self.assertFalse(company.uses_default_logo)
+
+    def create_companies(self):
+        env = self.env['res.company']
+        companies = [env.create({'name': f'company_{i}'}) for i in range(4)]
+        return env.browse([company.id for company in companies])
+
+    def test_setting_parent_company_for_single_child(self):
+        child_1, _, parent_1, parent_2 = self.create_companies()
+
+        # Can set parent company only for the first time
+        child_1.write({'parent_id': parent_1.id})
+
+        # The company hierarchy cannot be changed.
+        with self.assertRaises(UserError):
+            child_1.write({'parent_id': parent_2.id})
+
+    def test_setting_parent_company_for_multiple_children_01(self):
+        companies = self.create_companies()
+        children, parent_1, parent_2 = companies[:2], companies[2], companies[3]
+
+        # Can set parent company only for the first time
+        children.write({'parent_id': parent_1.id})
+
+        # The company hierarchy cannot be changed.
+        with self.assertRaises(UserError):
+            children.write({'parent_id': parent_2.id})
+
+    def test_setting_parent_company_for_multiple_children_02(self):
```

Comment:
instead of using two methods you could either do only one and not test this case (or if you really want this case of modifying only one children (but doesn't seem very useful), have it in the same test and have 3 children, and write on 2 on them one time, and on the 3rd one another time)